borderless/explainer.md

```diff
@@ -0,0 +1,512 @@
+      - [Disabling Window Management](#disabling-window-management)
+    - [5. Isolated Web App check](#5-isolated-web-app-check)
+  - [Related features](#related-features)
+    - [Displaying app’s origin in the app settings UI](#displaying-apps-origin-in-the-app-settings-ui)
+    - [Privacy indicators](#privacy-indicators)
+  - [Other notable details](#other-notable-details)
+    - [“Settings and more” three-dot button](#settings-and-more-three-dot-button)
+  - [Demo](#demo)
+    - [manifest.json](#manifestjson)
+    - [index.html](#indexhtml)
+    - [style.css](#stylecss)
+  - [Considered Alternatives](#considered-alternatives)
+    - [Having a separate permission for borderless](#having-a-separate-permission-for-borderless)
+    - [WCO extension](#wco-extension)
+      - [Reasonings for discarding](#reasonings-for-discarding)
+    - [New JS APIs](#new-js-apis)
+      - [Reasonings for discarding](#reasonings-for-discarding-1)
+  - [Security Considerations](#security-considerations)
+    - [Spoofing risks](#spoofing-risks)
+    - [Out-of-scope Navigation](#out-of-scope-navigation)
+    - [Opening a popup from a borderless PWA](#opening-a-popup-from-a-borderless-pwa)
+    - [Iframes](#iframes)
+
+## Introduction
+
+[VDI](https://www.softwaretestinghelp.com/best-vdi-software/#What_Is_Virtual_Desktop_Infrastructure)
+(Virtual Desktop Infrastructure) providers attempt to mirror remote applications
+to the local desktop. For example, the VDI provider's cloud is running a Text
+Processor (a desktop application window), which is to be presented on the client
+desktop as if the application was running locally.
+
+Installed web apps define their display mode in the manifest.json using display
+and
+[display_override](https://github.com/WICG/display-override/blob/main/explainer.md)
+fields. Currently
+[supported values for display modes](<https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/public/mojom/manifest/display_mode.mojom?q=-file:(test%7CDebug%7C%5C.xtb$)%20case:yes%20third_party%2Fblink%2Fpublic%2Fmojom%2Fmanifest%2Fdisplay_mode.mojom&ss=chromium>)
+are `window-controls-overlay`, `tabbed`, `minimal-ui`, `standalone`, `browser`
+and `fullscreen`. See
+[Manifest/display_override#values](https://developer.mozilla.org/en-US/docs/Web/Manifest/display_override#values)
+for more. From the before-mentioned options, `window-controls-overlay` has some
+same qualities to what we are trying to achieve. However, it is still not
+offering enough native-like experience for some of the use-cases and thus this
+explainer will explain how the title bar will be totally removed and enabling
+so-called borderless mode where the title bar area would be replaced with web
+content and so giving the developers full control on how the title bar would
+look like.
+
+## Example of a complex title bar with window-controls-overlay
+
+Some highly customized title bars would still look clumsy with
```

Comment:
This is a weird example for me, because it looks like exactly the sort of case that WCO is designed for. The screenshot shows WCO awkwardly overlapping the other content, but the solution to this (in the general case) is not to remove the native controls, but to properly use WCO's `titlebar-area` region to draw the other stuff in the correct place.

However, I guess this screenshot is intended to be showing a remote app in a VDI context that we have no control over, which means you cannot use the `titlebar-area` to customize the layout. So you should explain this: the use case is not a "highly customized title bar", it's a remote title bar that we have no control over, which already provides its own control buttons.